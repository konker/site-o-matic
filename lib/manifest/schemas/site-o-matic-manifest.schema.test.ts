/* eslint-disable @typescript-eslint/naming-convention */
import fs from 'node:fs';
import path from 'node:path';

import JSON5 from 'json5';

import * as unit from './site-o-matic-manifest.schema';

describe('site-o-matic-manifest-schema', () => {
  it('should validate a complete example', async () => {
    const json = await fs.promises.readFile(path.join(__dirname, '../../../docs/site-o-matic.manifest.example.json5'));
    const data = JSON5.parse(json.toString());
    const validation = unit.SiteOMaticManifest.parse(data);
    // expect(validation.success).toEqual(true);
    expect(validation).toMatchSnapshot();
  });

  it('should validate a minimal example', async () => {
    const json = await fs.promises.readFile(
      path.join(__dirname, '../../../docs/site-o-matic.manifest.example.minimal.json5')
    );
    const data = JSON5.parse(json.toString());
    const validation = unit.SiteOMaticManifest.parse(data);
    // expect(validation.success).toEqual(true);
    expect(validation).toStrictEqual({
      domainName: 'minimal-example.com',
      locked: false,
      webHosting: [
        {
          originPath: '/www',
          defaultRootObject: 'index.html',
          domainName: 'minimal-example.com',
          errorResponses: [
            {
              httpStatus: 403,
              responsePagePath: '/403.html',
            },
            {
              httpStatus: 404,
              responsePagePath: '/404.html',
            },
          ],
          type: 'cloudfront-s3',
          content: {
            producerId: 'default',
          },
        },
      ],
      webHostingDefaults: {
        'cloudfront-https': {
          originPath: '/',
        },
        'cloudfront-s3': {
          originPath: '/www',
          defaultRootObject: 'index.html',
          errorResponses: [
            {
              httpStatus: 403,
              responsePagePath: '/403.html',
            },
            {
              httpStatus: 404,
              responsePagePath: '/404.html',
            },
          ],
          content: {
            producerId: 'default',
          },
        },
        none: {},
      },
    });
  });
});
