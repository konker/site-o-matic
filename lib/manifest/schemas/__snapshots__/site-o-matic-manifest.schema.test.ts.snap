// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site-o-matic-manifest-schema should validate a complete example 1`] = `
{
  "domainName": "example.com",
  "extraDnsConfig": [
    {
      "hostName": "in1-smtp.example.com",
      "priority": 10,
      "type": "MX",
    },
    {
      "domainName": "fm1.example.com.dkim.example.com",
      "recordName": "fm1._domainkey.example.com",
      "type": "CNAME",
    },
    {
      "recordName": "",
      "type": "TXT",
      "values": [
        "v=spf1 include:spf.example.com ?all",
      ],
    },
  ],
  "locked": false,
  "notifications": {
    "noSubscription": false,
    "subscriptionEmail": "webmaster+{{context.manifest.rootDomainName}}@example.com",
  },
  "region": "eu-west-1",
  "registrar": "dynadot",
  "title": "example.com",
  "webHosting": [
    {
      "content": {
        "producerId": "default",
      },
      "defaultRootObject": "index.html",
      "domainName": "example.com",
      "errorResponses": [
        {
          "httpStatus": 403,
          "responsePagePath": "/403.html",
        },
        {
          "httpStatus": 404,
          "responsePagePath": "/404.html",
        },
      ],
      "originPath": "/main/www",
      "redirect": {
        "implementation": "edge-cf-function",
        "source": ".",
        "target": "https://www.example.org/",
      },
      "type": "cloudfront-s3",
    },
    {
      "content": {
        "producerId": "default",
      },
      "defaultRootObject": "index.html",
      "domainName": "www.example.com",
      "errorResponses": [
        {
          "httpStatus": 403,
          "responsePagePath": "/403.html",
        },
        {
          "httpStatus": 404,
          "responsePagePath": "/404.html",
        },
      ],
      "originPath": "/main/www",
      "type": "cloudfront-s3",
      "waf": {
        "AWSManagedRules": [
          {
            "name": "AWSManagedRulesCommonRuleSet",
            "priority": 10,
          },
          {
            "name": "AWSManagedRulesBotControlRuleSet",
            "priority": 20,
          },
        ],
        "enabled": true,
      },
    },
    {
      "auth": {
        "implementation": "edge-cf-function",
        "passwordSecretName": "dev.example.com_basic_auth_password",
        "type": "basic-auth",
        "usernameSecretName": "dev.example.com_basic_auth_username",
      },
      "content": {
        "producerId": "default",
      },
      "defaultRootObject": "index.html",
      "domainName": "dev.example.com",
      "errorResponses": [
        {
          "httpStatus": 403,
          "responsePagePath": "/403.html",
        },
        {
          "httpStatus": 404,
          "responsePagePath": "/404.html",
        },
      ],
      "originPath": "/dev/www",
      "type": "cloudfront-s3",
    },
    {
      "domainName": "api.bp-o-matic.development.konker.dev",
      "originPath": "/",
      "type": "cloudfront-https",
      "url": "https://8r3xuheel8.execute-api.eu-west-1.amazonaws.com/",
    },
  ],
  "webHostingDefaults": {
    "cloudfront-https": {
      "originPath": "/",
    },
    "cloudfront-s3": {
      "content": {
        "producerId": "default",
      },
      "defaultRootObject": "index.html",
      "errorResponses": [
        {
          "httpStatus": 403,
          "responsePagePath": "/403.html",
        },
        {
          "httpStatus": 404,
          "responsePagePath": "/404.html",
        },
      ],
      "originPath": "/www",
    },
    "none": {},
  },
  "webmasterEmail": "webmaster@example.com",
}
`;
